### ISSUE TITLE
Formatting new disks on GCE doesn't work: mkfs.auto not found

### ISSUE THREAD

**Issue by P-E-Meunier** [NixOS/nixops#755]
Since my setup of NixOps on GCE is relatively unstable, I tried to add a disk (managed by the Google interface, not by NixOps) so that I don't lose too much data.

Here are relevant portions of my deployment file:

```
  deployment.gce = credentials // {
    machineName = "blabla";
    region = "some-region";
    instanceType = "instance-type";
    tags = [ "http-server" "https-server" ];
    ipAddress = "my-ip";
    blockDeviceMapping."/dev/sdb" = { disk = "my-data"; };
  };
  fileSystems."/var/data" = {
    autoFormat = true;
    device = "/dev/sdb";
  };
```

The `autoFormat` option is not ignored, but the systemd service supposed to format that disk cannot find `mkfs.auto`. If I format it manually though, it gets mounted on reboot.

**Comment by aij** [comment]
Have you tried setting `fsType` to something other than the default? (`auto`)

Mount can guess the filesystem by looking at what's on the disk. Mkfs would have to read your mind to know what filesystem type you have in mind...

Perhaps nixos could add an assertion for when `autoFormat` is set with an unsupported `fsType`.

**Comment by P-E-Meunier** [comment]
Good point! More reasonably, this could be added to the manual (along with a small example).

**Comment by aij** [comment]
https://nixos.org/nixos/manual/options.html#opt-fileSystems._name__.autoFormat

Seems reasonably clear. I'm guessing that's not the documentation you were looking at?

**Comment by P-E-Meunier** [comment]
I meant added to the NixOps manual.

When I'm using NixOps, and the NixOps manual tells me it can deal with GCE disks, and NixOps already created a root image ("reading my mind" to decide what filesystem to use), I'm tempted to keep reading the NixOps manual, for instance https://nixos.org/nixops/manual/#idm140737316396832.

If this is not a serious UX flaw, I think this is at least an important NixOps documentation issue.

Examples would have easily prevented this problem.
